crates/pg_treesitter_queries/src/queries/relations.rs

```diff
@@ -0,0 +1,91 @@
+use crate::{Query, QueryResult};
+
+use super::QueryTryFrom;
+
+static QUERY: &'static str = r#"
+    (relation
+        (object_reference 
+            .
+            (identifier) @schema_or_table
+            "."?
+            (identifier)? @table
+        )+
+    )
+"#;
+
+#[derive(Debug)]
+pub struct RelationMatch<'a> {
+    pub(crate) schema: Option<tree_sitter::Node<'a>>,
+    pub(crate) table: tree_sitter::Node<'a>,
+}
+
+impl<'a> RelationMatch<'a> {
+    pub fn get_schema(&self, sql: &str) -> Option<String> {
+        let str = self
+            .schema
+            .as_ref()?
+            .utf8_text(sql.as_bytes())
+            .expect("Failed to get schema from RelationMatch");
+
+        Some(str.to_string())
+    }
+
+    pub fn get_table(&self, sql: &str) -> String {
+        self.table
+            .utf8_text(sql.as_bytes())
+            .expect("Failed to get schema from RelationMatch")
+            .to_string()
+    }
+}
+
+impl<'a> TryFrom<&'a QueryResult<'a>> for &'a RelationMatch<'a> {
+    type Error = String;
+
+    fn try_from(q: &'a QueryResult<'a>) -> Result<Self, Self::Error> {
+        match q {
+            QueryResult::Relation(r) => Ok(&r),
+
+            #[allow(unreachable_patterns)]
+            _ => Err("Invalid QueryResult type".into()),
+        }
+    }
+}
+
+impl<'a> QueryTryFrom<'a> for RelationMatch<'a> {
+    type Ref = &'a RelationMatch<'a>;
+}
+
+impl<'a> Query<'a> for RelationMatch<'a> {
+    fn execute(root_node: tree_sitter::Node<'a>, stmt: &'a str) -> Vec<crate::QueryResult<'a>> {
+        let query =
+            tree_sitter::Query::new(tree_sitter_sql::language(), &QUERY).expect("Invalid Query.");
```

Comment:
I tried using a `tokio::sync::OnceCell` here so we would only parse the query once.. but that led to an insurmountable amount of pain and frustration (the `Workspace` trait had to be rewritten)

I would be open to use a `thread_local` here, so at least we'd only have a query per threat. WDYT?

Reply:
what about a std::sync::LazyLock? but I dont know whether it matter a lot in terms of performance?

Reply:
Hmm I tried it, and the tests seem to pass. I worry a bit about what happens when we have many concurrent threads, but I guess there'll rarely be more than one autocompletion request, so let's assume it works :)